src/index.js

```diff
@@ -179,6 +179,20 @@ export default class ScrollBehavior {
     scrollTop(element, y);
   }
 
+  _defaultScrollTarget(hash) {
+    if (hash && hash !== '#') {
+      const id = hash.at(0) === '#' ? hash.substring(1) : hash;
+      const el = document.getElementById(id) ||
+        document.getElementsByName(id)[0];
```

Comment:
Back in the day the way URL fragments were used was via something like `<a name="foo">` to mark the place on the page where `#foo` should go. That is now deprecated in favor of `id`, but browsers still support it.

In fact, we currently have JSX code that still uses it because it was copy-pasted from an old XHTML page... I'll probably be fixing that shortly.

Here's the relevant section of HTML 4.0: https://www.w3.org/TR/html401/struct/links.html#h-12.2.1 (Which happens to be another example that uses the `name` attribute. :)